modules/core/server/controllers/core.server.controller.js

```diff
@@ -94,3 +94,27 @@ exports.renderServiceWorkerConfig = function (req, res) {
     .set('Content-Type', 'text/javascript')
     .send('var FCM_SENDER_ID = ' + JSON.stringify(config.fcm.senderId) + ';\n');
 };
+
+/**
+ * Format languages from object format into array compatible with React component used in UI
+ */
+function languageObjectToArray(languages) {
+  const langsArr = [];
+
+  Object.entries(languages).forEach(language => {
+    langsArr.push({ value: language[0], label: language[1] });
+  });
+
+  return langsArr;
+}
+
+exports.getLanguages = (req, res) => {
+  let languages = require(path.resolve('config/languages/languages.json'));
+
+  // Reformat object into array if so requested
+  if (req?.query?.format === 'array') {
+    languages = languageObjectToArray(languages);
```

Comment:
Todo; instead of doing this on the fly with the server, how about we do this at compile-time, store the file at the server and simply read it?